ACPs/196-precompile-secp256r1/README.md

```diff
@@ -0,0 +1,83 @@
+- **Cost optimization**: 100x gas reduction (from 200k-330k to 3,450 gas) makes enterprise-scale applications economically viable
+
+The 100x gas cost reduction makes these use cases economically viable while maintaining the security properties institutions and users expect from their existing devices.
+
+## Specification
+
+This ACP implements [EIP-7212](https://eips.ethereum.org/EIPS/eip-7212) for secp256r1 signature verification on Avalanche. The specification follows EIP-7212 exactly, with the precompiled contract deployed at address `0x0000000000000000000000000000000000000100`.
+
+### Core Functionality
+- **Input**: 160 bytes (message hash + signature components r,s + public key coordinates x,y)
+- **Output**: 32 bytes (success: `0x...01`, failure: empty)  
+- **Gas Cost**: 3,450 gas (based on EIP-7212 benchmarking)
+- **Validation**: Full compliance with NIST FIPS 186-5 specification
+
+### Activation
+This precompile may be activated as part of Avalanche's next network upgrade. Individual Avalanche L1s and subnets could adopt this enhancement independently through their respective client software updates.
+
+For complete technical specifications, validation requirements, and implementation details, refer to [EIP-7212](https://eips.ethereum.org/EIPS/eip-7212).
+
+## Backwards Compatibility
+
+This ACP introduces a new precompiled contract and does not modify existing functionality. No backwards compatibility issues are expected since:
+
+1. The precompile uses a previously unused address (`0x100`)
+2. No existing opcodes or consensus rules are modified
+3. The change is additive and opt-in for applications
+
+Adoption requires a coordinated network upgrade for the C-Chain. Subnet-EVM chains can adopt this enhancement independently by upgrading their client software.
+
+## Security Considerations
+
+### Cryptographic Security
+- The secp256r1 curve is standardized by NIST and widely vetted
+- Security properties are comparable to secp256k1 (used by ECRECOVER)
+- Implementation follows NIST FIPS 186-5 specification exactly
+
+### Implementation Security  
+- Signature verification (not recovery) approach maximizes compatibility with existing P-256 ecosystem
+- No malleability check included to match NIST specification, but wrapper libraries should add this
+- Input validation prevents invalid curve points and out-of-range signature components
+
+### Network Security
+- Gas cost prevents potential DoS attacks through expensive computation
+- Precompile implementation should use battle-tested cryptographic libraries
+- No consensus-level security implications beyond standard precompile considerations
+
+## Reference Implementation
+
+The implementation will build upon existing work:
+
+1. **EIP-7212 Reference**: The [go-ethereum implementation](https://github.com/ethereum/go-ethereum) of EIP-7212 provides the foundation
```

Comment:
I can't find an implementation anywhere in their repo.

Reply:
[Here](https://github.com/maticnetwork/bor/pull/1069) it is. Also, updating the link.